audio-echo/app/src/main/java/com/google/sample/echo/MainActivity.java

```diff
@@ -38,6 +46,8 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        //get run-time permission...
+        requestResourcePermission();
         status_view = (TextView)findViewById(R.id.statusView);
         queryNativeAudioParameters();
```

Comment:
Yes, will follow the samples JF recommended and compare the implementation.  The necessary idea for this sample is to protect "start" button: when "Start button" is clicked, if have permission --> call into JNI; if no permission, do nothing [ definitely not calling into JNI ] to avoid crash